### PR TITLE
[tests-only] change collaborator permissions

### PR DESCRIPTION
**Pull request by swoichha**
## Description
This PR adds test for changing collaborator permissions got files/folder.

## Related Issue
https://github.com/owncloud/client/issues/8651

### CODE REVIEW

File: test/gui/shared/scripts/names.py [hunk 1]
```diff
@@ -98,3 +98,4 @@
 o_tableView_0_1_QModelIndex = {"column": 1, "container": oCC_IssuesWidget_tableView_QTableView, "row": 0, "type": "QModelIndex"}
 settings_settingsdialog_toolbutton_Add_account_QToolButton = {"name": "settingsdialog_toolbutton_Add account", "type": "QToolButton", "visible": 1, "window": settings_OCC_SettingsDialog}
 settings_settingsdialog_toolbutton_Activity_QToolButton = {"name": "settingsdialog_toolbutton_Activity", "type": "QToolButton", "visible": 1, "window": settings_OCC_SettingsDialog}
+scrollArea_permissionToolButton_QToolButton = {"container": sharingDialogUG_scrollArea_QScrollArea, "name": "permissionToolButton", "type": "QToolButton", "visible": 1}
```
Comment (Talank) [inline test/gui/shared/scripts/names.py:101]: I think this selector is not needed.

File: test/gui/shared/scripts/pageObjects/SharingDialog.py [hunk 2]
```diff
@@ -82,3 +88,21 @@ def addCollaborator(self, receiver, permissions):
 
     def getErrorText(self):
         return str(squish.waitForObjectExists(self.ERROR_SHOWN_ON_SHARING_DIALOG).text)
+
+    def removePermissions(self, receiver, permissions):
+        removePermissionsList = permissions.split(",")
+        print(removePermissionsList)
```
Comment (Talank) [inline test/gui/shared/scripts/pageObjects/SharingDialog.py:94]: this print seems to be a debug line

File: test/gui/shared/scripts/pageObjects/SharingDialog.py [hunk 3]
```diff
@@ -82,3 +88,20 @@ def addCollaborator(self, receiver, permissions):
...
         return str(squish.waitForObjectExists(self.ERROR_SHOWN_ON_SHARING_DIALOG).text)
+
+    def removePermissions(self, receiver, permissions):
+        removePermissionsList = permissions.split(",")
+        (
+            isEditPermissionAvailable,
+            isSharePermissionAvailable,
+        ) = self.getAvailablePermission()
+
+        if 'share' in removePermissionsList and isSharePermissionAvailable == True:
```
Comment (Talank) [inline test/gui/shared/scripts/pageObjects/SharingDialog.py:99]:
```suggestion
        if 'share' in removePermissionsList and isSharePermissionAvailable:
```
We do not need to compare if the variable is already a boolean.

File: test/gui/shared/scripts/pageObjects/SharingDialog.py [hunk 4]
```diff
@@ -82,3 +88,20 @@ def addCollaborator(self, receiver, permissions):
...
+            squish.clickButton(
+                squish.waitForObject(names.scrollArea_permissionShare_QCheckBox)
+            )
+
+        if 'edit' in removePermissionsList and isEditPermissionAvailable == True:
```
Comment (Talank) [inline test/gui/shared/scripts/pageObjects/SharingDialog.py:104]:
```suggestion
        if 'edit' in removePermissionsList and isEditPermissionAvailable:
```
Same here

**Review by Talank — COMMENTED** [review]
just one suggestion.
All in all, it looks good to me

**Review by Talank — COMMENTED** [review]
lgtm

File: test/gui/shared/steps/steps.py [hunk 5]
```diff
@@ -646,3 +646,35 @@ def step(context):
...
+    shareItem = SharingDialog()
+    if 'edit' in permissionsList:
+        test.compare(
+            waitForObjectExists(shareItem.EDIT_PERMISSIONS_CHECKBOX).checked, False
+        )
+
+    if 'share' in permissionsList:
+        test.compare(
+            waitForObjectExists(shareItem.SHARE_PERMISSIONS_CHECKBOX).checked, False
+        )
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:680]: can we use `getAvailablePermission` here?

File: test/gui/tst_sharing/test.feature [hunk 6]
```diff
@@ -134,3 +134,22 @@ Feature: Sharing
...
+        Given user "Alice" has created on the server folder "simple-folder"
+        And user "Alice" on the server has created file "lorem.txt"
+        And user "Brian" has been created on the server with default attributes and without skeleton files
+        And user "Alice" has shared folder "simple-folder" on the server with user "Brian"
+        And user "Alice" has shared folder "lorem.txt" on the server with user "Brian"
+        And user "Alice" has set up a client with default settings
+        When the user removes permissions "<permissions>" for user "Brian Murphy" of resource "%client_sync_path%/simple-folder" using the client-UI
+        And the user removes permissions "<permissions>" for user "Brian Murphy" of resource "%client_sync_path%/lorem.txt" using the client-UI
+        Then the user should not have "<permissions>" permissions for user "Brian Murphy" for resource "%client_sync_path%/simple-folder"
+        And the user should not have "<permissions>" permissions for user "Brian Murphy" for resource "%client_sync_path%/lorem.txt"
```
Comment (individual-it) [inline test/gui/tst_sharing/test.feature:149]: I think we should check through the API if those permissions really got changed on the server

File: test/gui/tst_sharing/test.feature [hunk 7]
```diff
@@ -134,3 +134,22 @@ Feature: Sharing
...
             | name        | Public link    |
         And the public should not be able to download the file "lorem.txt" from the last created public link by "Alice" on the server
+
+
+    Scenario Outline: change collaborator permissions of a file & folder
+        Given user "Alice" has created on the server folder "simple-folder"
+        And user "Alice" on the server has created file "lorem.txt"
+        And user "Brian" has been created on the server with default attributes and without skeleton files
+        And user "Alice" has shared folder "simple-folder" on the server with user "Brian"
+        And user "Alice" has shared folder "lorem.txt" on the server with user "Brian"
```
Comment (individual-it) [inline test/gui/tst_sharing/test.feature:144]: it seems to be a bit random where we put in `on the server`. What do you think @phil-davis ? How do those sentences sound for your?

**Review by individual-it — APPROVED** [review]

**Review by individual-it — CHANGES_REQUESTED** [review]
some small comments

File: test/gui/tst_sharing/test.feature (hunk 7 above)
Comment (phil-davis) [reply test/gui/tst_sharing/test.feature]: `on the server` is better at the end:
`        Given user "Alice" has created folder "simple-folder" on the server`

`        Given user "Alice" has created file "lorem.txt" on the server`

The `with/without` clauses sound OK after "on the server". So the pattern is:

```
Given user "Username" has [created|deleted|shared|...] [some resource] on the server (with some special stuff|with permission|with email address|...)
```

File: test/gui/tst_sharing/test.feature (hunk 6 above)
Comment (individual-it) [reply test/gui/tst_sharing/test.feature]: Sorry, my comment was not clear.
I didn't mean to delete those UI test steps, but check additionally with the API
Maybe we can also rephrase those UI steps:
`Then "<permissions>" permissions should not be displayed for user "Brian Murphy" for resource "%client_sync_path%/simple-folder on the UI"`

File: test/gui/tst_sharing/test.feature [hunk 8]
```diff
@@ -140,16 +140,33 @@ Feature: Sharing
         Given user "Alice" has created on the server folder "simple-folder"
         And user "Alice" on the server has created file "lorem.txt"
         And user "Brian" has been created on the server with default attributes and without skeleton files
-        And user "Alice" has shared folder "simple-folder" on the server with user "Brian"
-        And user "Alice" has shared folder "lorem.txt" on the server with user "Brian"
+    	And user "Alice" on the server has shared folder "simple-folder" with user "Brian" with "all" permissions
+    	And user "Alice" on the server has shared file "lorem.txt" with user "Brian" with "all" permissions
         And user "Alice" has set up a client with default settings
         When the user removes permissions "<permissions>" for user "Brian Murphy" of resource "%client_sync_path%/simple-folder" using the client-UI
+        And the user closes the sharing dialog
```
Comment (individual-it) [inline test/gui/tst_sharing/test.feature]: indents seem to be a bit messed up, maybe your IDE gets confused with tabs and spaces?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/tst_sharing/test.feature (hunk 8 above)
Comment (swoichha) [reply test/gui/tst_sharing/test.feature]: fixed

File: test/gui/tst_sharing/test.feature (hunk 7 above)
Comment (swoichha) [reply test/gui/tst_sharing/test.feature]: @phil-davis @individual-it This sounds good. The required step for this is already in progress here https://github.com/owncloud/client/pull/8695/files#diff-821cad5db62830778c291de069b5182ff9bcdd90951faa2cec9d223e8efbcc20R294. We can make this change in another PR where all `Given` steps can be refactored.

**Review by individual-it — APPROVED** [review]

**Pull request merged**